src/components/error-overlay/error-message-id-provider.ts

```diff
@@ -0,0 +1,33 @@
+type ErrorDetails = {
+  title: string;
+  message: string;
+};
+
+export const getErrorMessageIdByCategory = (errorCategory: number | undefined): ErrorDetails => {
```

Comment:
i think a key-value map/object of code => error details would be cleaner.
just something like errorMap.get(code) || {...//default value}